File: 0401-binary-watch/0401-binary-watch.py
def countBits(n: int) -> int:
    bits = 0
    mask = 1
    for i in range(10):
        if mask & n:
            bits += 1
        mask <<= 1
    return bits
class Solution:
    def readBinaryWatch(self, turnedOn: int) -> List[str]:
        ans = []
        hoursMask = 960
        minsMask = 63
        for i in range(1, 1025):
            if countBits(i) == turnedOn:
                h = (hoursMask & i) >> 6
                m = minsMask & i
                if h > 11 or m > 59:
                    continue
                ans.append(f"{h}:{m}") if m >= 10 else ans.append(f"{h}:0{m}")
        return ans
        

File: 0401-binary-watch/0401-binary-watch.cpp
class Solution {
public:
    vector<string> readBinaryWatch(int turnedOn) {
        vector<string> res;
        int hoursMask = 960;
        int minsMask = 63;
        for(int i = 1; i <= 1024; i++)
        {
            if(countBits(i) == turnedOn)
            {
                int h = (hoursMask & i) >> 6;
                int m = minsMask & i;
                if(h > 11 || m > 59)
                {
                    continue;
                }
                res.push_back(to_string(h) + ":" + (m < 10 ? "0" : "") + to_string(m));
            }
        }
        return res;
    }
private:
    char countBits(int n)
    {
        char bits = 0;
        int mask = 1;
        for(int i = 0; i < 10; i++)
        {
            if(mask & n)
            {
                bits++;
            }
            mask <<= 1;
        }
        return bits;
    }
};